packages/webpack/react-webpack-plugin/src/ReactWebpackPlugin.ts

```diff
@@ -224,6 +249,106 @@ class ReactWebpackPlugin {
             });
         },
       );
+      if (options.extractStr) {
+        compilation.hooks.processAssets.tap(
+          {
+            name: 'compilation',
+            stage:
+              compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE,
+          },
+          (assets) => {
+            const entryIdx2selectStrVec: Record<
+              number,
+              string[]
+            > = {};
+            const DEFAULT_STR_LENGTH = 20;
+            const jsAssets = Object.keys(assets).filter(name =>
+              name.endsWith('.js')
+            );
+            jsAssets.forEach(
+              (key) => {
+                const entryIdx = options.mainThreadChunks?.findIndex(
+                  (chunkName) => key.includes(chunkName),
+                );
+                if (entryIdx !== undefined && entryIdx !== -1) {
+                  const result = transformBundleResultSync(
+                    assets[key]!.source().toString(),
+                    {
+                      filename: key,
+                      pluginName: 'transformBundleResult',
```

Comment:
Can we use a different `pluginName` from the one used in the background? This way, if there are errors or warnings, we can differentiate them through the error messages.